From b933a3c337dbbe383fd63ae5f06cd53d25db05f3 Mon Sep 17 00:00:00 2001
From: Edison Ai <edison.ai@arm.com>
Date: Thu, 24 May 2018 11:18:07 +0800
Subject: [PATCH 09/36] libmbedtls: configure mbedTLS for different modes

Split mbedTLS into 3 partitions: CRYPTO, X509 and TLS. CRYPTO is for
kernel and user mode. X509 and TLS are mainly for user mode.

Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Edison Ai <edison.ai@arm.com>
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>
---
 lib/libmbedtls/sub.mk | 153 ++++++++++++++++++++++--------------------
 1 file changed, 79 insertions(+), 74 deletions(-)

diff --git a/lib/libmbedtls/sub.mk b/lib/libmbedtls/sub.mk
index 6f1fcfbc..01897404 100644
--- a/lib/libmbedtls/sub.mk
+++ b/lib/libmbedtls/sub.mk
@@ -1,87 +1,92 @@
 global-incdirs-y += include
 global-incdirs-y += mbedtls/include
 
-SRCS :=
-
 # OBJS_CRYPTO from make file
-SRCS += aes.c
-SRCS += aesni.c
-SRCS += arc4.c
-SRCS += asn1parse.c
-SRCS += asn1write.c
-SRCS += base64.c
-SRCS += bignum.c
-SRCS += blowfish.c
-SRCS += camellia.c
-SRCS += ccm.c
-SRCS += cipher.c
-SRCS += cipher_wrap.c
-SRCS += cmac.c
-SRCS += ctr_drbg.c
-SRCS += des.c
-SRCS += dhm.c
-SRCS += ecdh.c
-SRCS += ecdsa.c
-SRCS += ecjpake.c
-SRCS += ecp.c
-SRCS += ecp_curves.c
-SRCS += entropy.c
-SRCS += entropy_poll.c
-SRCS += error.c
-SRCS += gcm.c
-SRCS += havege.c
-SRCS += hmac_drbg.c
-SRCS += md.c
-SRCS += md2.c
-SRCS += md4.c
-SRCS += md5.c
-SRCS += md_wrap.c
-SRCS += memory_buffer_alloc.c
-SRCS += oid.c
-SRCS += padlock.c
-SRCS += pem.c
-SRCS += pk.c
-SRCS += pk_wrap.c
-SRCS += pkcs12.c
-SRCS += pkcs5.c
-SRCS += pkparse.c
-SRCS += pkwrite.c
-SRCS += platform.c
-SRCS += ripemd160.c
-SRCS += rsa.c
-SRCS += sha1.c
-SRCS += sha256.c
-SRCS += sha512.c
-SRCS += threading.c
-SRCS += timing.c
-SRCS += version.c
-SRCS += version_features.c
-SRCS += xtea.c
+SRCS_CRYPTO :=
+SRCS_CRYPTO += aes.c
+SRCS_CRYPTO += aesni.c
+SRCS_CRYPTO += arc4.c
+SRCS_CRYPTO += asn1parse.c
+SRCS_CRYPTO += asn1write.c
+SRCS_CRYPTO += base64.c
+SRCS_CRYPTO += bignum.c
+SRCS_CRYPTO += blowfish.c
+SRCS_CRYPTO += camellia.c
+SRCS_CRYPTO += ccm.c
+SRCS_CRYPTO += cipher.c
+SRCS_CRYPTO += cipher_wrap.c
+SRCS_CRYPTO += cmac.c
+SRCS_CRYPTO += ctr_drbg.c
+SRCS_CRYPTO += des.c
+SRCS_CRYPTO += dhm.c
+SRCS_CRYPTO += ecdh.c
+SRCS_CRYPTO += ecdsa.c
+SRCS_CRYPTO += ecjpake.c
+SRCS_CRYPTO += ecp.c
+SRCS_CRYPTO += ecp_curves.c
+SRCS_CRYPTO += entropy.c
+SRCS_CRYPTO += entropy_poll.c
+SRCS_CRYPTO += error.c
+SRCS_CRYPTO += gcm.c
+SRCS_CRYPTO += havege.c
+SRCS_CRYPTO += hmac_drbg.c
+SRCS_CRYPTO += md.c
+SRCS_CRYPTO += md2.c
+SRCS_CRYPTO += md4.c
+SRCS_CRYPTO += md5.c
+SRCS_CRYPTO += md_wrap.c
+SRCS_CRYPTO += memory_buffer_alloc.c
+SRCS_CRYPTO += oid.c
+SRCS_CRYPTO += padlock.c
+SRCS_CRYPTO += pem.c
+SRCS_CRYPTO += pk.c
+SRCS_CRYPTO += pk_wrap.c
+SRCS_CRYPTO += pkcs12.c
+SRCS_CRYPTO += pkcs5.c
+SRCS_CRYPTO += pkparse.c
+SRCS_CRYPTO += pkwrite.c
+SRCS_CRYPTO += platform.c
+SRCS_CRYPTO += ripemd160.c
+SRCS_CRYPTO += rsa.c
+SRCS_CRYPTO += sha1.c
+SRCS_CRYPTO += sha256.c
+SRCS_CRYPTO += sha512.c
+SRCS_CRYPTO += threading.c
+SRCS_CRYPTO += timing.c
+SRCS_CRYPTO += version.c
+SRCS_CRYPTO += version_features.c
+SRCS_CRYPTO += xtea.c
 
 # OBJS_X509
-SRCS += certs.c
-SRCS += pkcs11.c
-SRCS += x509.c
-SRCS += x509_create.c
-SRCS += x509_crl.c
-SRCS += x509_crt.c
-SRCS += x509_csr.c
-SRCS += x509write_crt.c
-SRCS += x509write_csr.c
+SRCS_X509 :=
+SRCS_X509 += certs.c
+SRCS_X509 += pkcs11.c
+SRCS_X509 += x509.c
+SRCS_X509 += x509_create.c
+SRCS_X509 += x509_crl.c
+SRCS_X509 += x509_crt.c
+SRCS_X509 += x509_csr.c
+SRCS_X509 += x509write_crt.c
+SRCS_X509 += x509write_csr.c
 
 # OBJS_TLS
-SRCS += debug.c
-SRCS += net_sockets.c
-SRCS += ssl_cache.c
-SRCS += ssl_ciphersuites.c
-SRCS += ssl_cli.c
-SRCS += ssl_cookie.c
-SRCS += ssl_srv.c
-SRCS += ssl_ticket.c
-SRCS += ssl_tls.c
+SRCS_TLS :=
+SRCS_TLS += debug.c
+SRCS_TLS += net_sockets.c
+SRCS_TLS += ssl_cache.c
+SRCS_TLS += ssl_ciphersuites.c
+SRCS_TLS += ssl_cli.c
+SRCS_TLS += ssl_cookie.c
+SRCS_TLS += ssl_srv.c
+SRCS_TLS += ssl_ticket.c
+SRCS_TLS += ssl_tls.c
 
-srcs-y += $(addprefix mbedtls/library/, $(SRCS))
+srcs-y += $(addprefix mbedtls/library/, $(SRCS_CRYPTO))
+srcs-$(sm-$(ta-target)) += $(addprefix mbedtls/library/, $(SRCS_X509))
+srcs-$(sm-$(ta-target)) += $(addprefix mbedtls/library/, $(SRCS_TLS))
 
 cflags-lib-y += -Wno-redundant-decls
 cflags-lib-y += -Wno-switch-default
+ifneq ($(sm),core) # User-mode
 cflags-lib-$(CFG_ULIBS_GPROF) += -pg
+endif
-- 
2.17.1

